test/e2e/placement_pickfixed_test.go

```diff
@@ -0,0 +1,179 @@
+						ClusterNames: []string{
+							memberCluster1Name,
+						},
+					},
+				},
+			}
+			Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP")
+
+			// Verify that resources are placed on specified clusters.
+			resourcePlacedActual := workNamespaceAndConfigMapPlacedOnClusterActual(memberCluster1)
+			Eventually(resourcePlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place resources on specified clusters")
+
+			// Update the CRP to pick a different cluster.
+			Expect(hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp)).To(Succeed(), "Failed to get CRP")
+			crp.Spec.Policy.ClusterNames = []string{memberCluster2Name}
+			Expect(hubClient.Update(ctx, crp)).To(Succeed(), "Failed to update CRP")
+		})
+
+		It("should place resources on newly specified clusters", func() {
+			resourcePlacedActual := workNamespaceAndConfigMapPlacedOnClusterActual(memberCluster2)
+			Eventually(resourcePlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place resources on specified clusters")
+		})
+
+		It("should remove resources from previously specified clusters", func() {
+			resourceRemovedActual := workNamespaceRemovedFromClusterActual(memberCluster1)
+			Eventually(resourceRemovedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to remove resources from previously specified clusters")
+		})
+
+		It("should update CRP status as expected", func() {
+			crpStatusUpdatedActual := crpStatusUpdatedActual(workResourceIdentifiers(), []string{memberCluster2Name}, nil)
+			Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP status as expected")
+		})
+
+		AfterAll(func() {
+			ensureCRPAndRelatedResourcesDeletion(crpName, []*framework.Cluster{memberCluster2})
+		})
+	})
+
+	Context("pick unhealthy and non-existent clusters", Ordered, func() {
+		crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+
+		BeforeAll(func() {
+			// Create the resources.
+			createWorkResources()
+
+			// Create the CRP.
+			crp := &placementv1beta1.ClusterResourcePlacement{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: crpName,
+					// Add a custom finalizer; this would allow us to better observe
+					// the behavior of the controllers.
+					Finalizers: []string{customDeletionBlockerFinalizer},
+				},
+				Spec: placementv1beta1.ClusterResourcePlacementSpec{
+					ResourceSelectors: workResourceSelector(),
+					Strategy: placementv1beta1.RolloutStrategy{
+						Type: placementv1beta1.RollingUpdateRolloutStrategyType,
+						RollingUpdate: &placementv1beta1.RollingUpdateConfig{
+							UnavailablePeriodSeconds: pointer.Int(2),
+						},
+					},
+					Policy: &placementv1beta1.PlacementPolicy{
+						PlacementType: placementv1beta1.PickFixedPlacementType,
+						ClusterNames: []string{
+							memberCluster4Name,
+							memberCluster5Name,
+							memberCluster6Name,
```

Comment:
I guess some of them are unhealthy cluster, it would be great if their name reflect that to improve the test readability.